src/main/java/DemandeAvenantComponent.java

```diff
@@ -0,0 +1,200 @@
+				if (!garantieContrat.getCouvertures().isEmpty()) {
+					CouverturePivot couverture = garantieDemandeAvenantAcceptee.getCouvertures().get(0);
+					BigDecimal montantTotal = getMontantTotal(garantieContrat, garantieDemandeAvenantAcceptee);
+					couverture.setMontantAssure(montantTotal);
+					garantieDemandeAvenantAcceptee.setCouvertures(Collections.singletonList(couverture));
+				}
+			}
+		}
+	}
+
+	private GarantiePivot getGarantieDemandeAvenantAcceptee(DossierPivot dossier, GarantiePivot garantieContrat) {
+		return dossier.getContrat().getInformations().getGaranties().stream()
+				.filter(g -> g.getCodeCombinaisonFils() != null && g.getCodeCombinaisonFils().getCodePlan().equals(garantieContrat.getCodeCombinaisonFils().getCodePlan())).findFirst().orElse(null);
+	}
+
+	private BigDecimal getMontantTotal(GarantiePivot garantieContrat, GarantiePivot garantieDemandeAvenant) {
+		List<CouverturePivot> couvertures = montantAssureDemandeAvenantEditeeInferieurAMontantAsurreContrat(garantieContrat, garantieDemandeAvenant) ? garantieDemandeAvenant.getCouvertures()
+				: garantieContrat.getCouvertures();
+		return couvertures.stream().filter(nePasPrendreLesCouverturesRefusees()).map(c -> c.getMontantAssure() != null ? c.getMontantAssure() : BigDecimal.ZERO).reduce(BigDecimal.ZERO,
+				BigDecimal::add);
+	}
+
+	private boolean montantAssureDemandeAvenantEditeeInferieurAMontantAsurreContrat(GarantiePivot garantieContrat, GarantiePivot garantieDemandeAvenant) {
+		List<CouverturePivot> couverturesContrat = garantieContrat.getCouvertures();
+		List<CouverturePivot> couverturesDemandeAvenant = garantieDemandeAvenant.getCouvertures();
+
+		if (!couverturesContrat.isEmpty() && !couverturesDemandeAvenant.isEmpty()) {
+			BigDecimal montantAssureContrat = couverturesContrat.get(0).getMontantAssure();
+			BigDecimal montantAssureDemandeAvenant = couverturesDemandeAvenant.get(0).getMontantAssure();
+
+			if (montantAssureContrat != null && montantAssureDemandeAvenant != null) {
+				return montantAssureDemandeAvenant.compareTo(montantAssureContrat) < 0;
+			}
+		}
+
+		return false;
+	}
+
+	public void suspendreDemandeAvenant(final BigInteger idDemandeAvenant) {
+		sDemandeAvenant.suspendreDemandeAvenant(idDemandeAvenant);
+	}
+
+	public void enregistrerTarif(Optional<DemandeAvenantPivot> demandeAvenant) {
+		sDemandeAvenant.enregistrerTarif(demandeAvenant);
+	}
+
+	public Optional<Boolean> enregistrerNumeroTravail(BigInteger idDemandeAvenant, String numeroTravail) {
+		if (idDemandeAvenant != null) {
+			Optional<DemandeAvenantEntity> oDemandeAvenant = repositoryDemandeAvenant.findById(idDemandeAvenant);
+			if (oDemandeAvenant.isPresent()) {
+				oDemandeAvenant.get().setNumeroTravail(numeroTravail);
+				repositoryDemandeAvenant.save(oDemandeAvenant.get());
+				return Optional.of(Boolean.TRUE);
+			}
+		}
+		return Optional.of(Boolean.FALSE);
+	}
+
+	public Optional<DemandeAvenantPivot> getDemandeAvenantAcceptee(String refContrat) {
```

Comment:
Method is responsible for getting an accepted demand based on a contract reference, violating SRP. (Severity: low)